PowerUp/app/src/main/java/powerup/systers/com/kill_the_virus_game/KillTheVirusGame.java

```diff
@@ -30,49 +42,237 @@
+            case PowerUpUtils.VIRUS_8:
+                imgVirus8.setVisibility(View.GONE);
+                break;
+            default:
+                imgVirus1.setVisibility(View.GONE);
+                break;
+        }
+        score = increasePoint(score);
+        txtScore.setText(Integer.toString(score));
+        if (score == 8)
+            gameEnd(score);
     }
 
-    public void initialSetup(){
-        //TODO: Initialize the views at the start
+    /**
+     * End the game if the hit was incorrect
+     */
+    public void wrongHit() {
+        correctHit = false;
+        gameEnd(score);
     }
 
-    public void rightHit(){
-        //TODO: Increase the value of score by one
-        //TODO: The Virus should disappear
+    /**
+     * Increases the points for correct hit
+     * @param currentScore - the current score
+     * @return - increases the value of current score by one
+     */
+    public int increasePoint(int currentScore) {
+        return currentScore + 1;
     }
 
-    public void wrongHit(){
-        //TODO: end the game
+    /**
+     * Saves the total score and ends the game
+     * @param totalScore - value of total score
+     */
+    private void gameEnd(int totalScore) {
+        //Update the total points
+        SharedPreferences sharedPref = this.getPreferences(Context.MODE_PRIVATE);
+        SharedPreferences.Editor editor = sharedPref.edit();
+        editor.putInt(getString(R.string.score_kill_the_virus), totalScore);
+        editor.apply();
+        //End the Game
+        finish();
     }
 
-    public int increasePoint(int currentScore){
-        return currentScore +1;
+    //Attaches rotational animation to all the virus images
+    private ValueAnimator animateVirus(final ImageView virus, long duration, int startPoint, int endPoint) {
+        ValueAnimator anim = ValueAnimator.ofInt(startPoint, endPoint);
+        anim.addUpdateListener(new ValueAnimator.AnimatorUpdateListener() {
+            @Override
+            public void onAnimationUpdate(ValueAnimator valueAnimator) {
+                int val = (Integer) valueAnimator.getAnimatedValue();
+                ConstraintLayout.LayoutParams layoutParams = (ConstraintLayout.LayoutParams) virus.getLayoutParams();
+                layoutParams.circleAngle = val;
+                virus.setLayoutParams(layoutParams);
+            }
+        });
+        anim.setDuration(duration);
+        anim.setInterpolator(new LinearInterpolator());
+        anim.setRepeatMode(ValueAnimator.RESTART);
+        anim.setRepeatCount(ValueAnimator.INFINITE);
+        return anim;
     }
 
-    public void gameEnd(){
-        //TODO: Update the total points
-        //TODO: Migrate to game completion screen
+    /**
+     * Matches the color of the virus with syringe and calls rightHit() or wrongHit() accordingly
+     * @param virusType - the type of virus that was shot ( 0-Purple, 1-Green, 2-Pink, 3-Black )
+     */
+    private void checkHit(int virusType) {
```

Comment:
I need help to resolve this.

Reply:
It's because of the multiple if-else statements in the code, which gives a high number of possible paths. to resolve this, either have to simplify the code or think of a different approach

Reply:
@thisum Can you please suggest ways on how this could be simplified

Reply:
You can do something like:
        if (virusType == PowerUpUtils.VIRUS_TYPE_1 && (int) virus1Animator.getAnimatedValue() >= 70 && (int) virus1Animator.getAnimatedValue() <= 110) {
            rightHit(PowerUpUtils.VIRUS_1);
        }
        else if (virusType == PowerUpUtils.VIRUS_TYPE_2 && (int) virus2Animator.getAnimatedValue() >= 70 && (int) virus2Animator.getAnimatedValue() <= 110) {
            rightHit(PowerUpUtils.VIRUS_2);
        }
........
else{
wrongHit();
}